index.html

```diff
@@ -0,0 +1,475 @@
+
+This illusion is maintained by the router, which must break incoming queries into subqueries. In the example above, the path `rockets.captain` is resolved by the `rockets::Rocket.astronaut` resolver, so its selection set is realized as a `rockets::Astronaut`. However, there is no `name` field for `rockets::Astronaut`—instead, the only resolver for `Astronaut.name` is `astronauts::Astronaut.name`. To process this query, the router must "convert" a `rockets::Astronaut` to an `astronauts::Astronaut` via an `_entities` query.
+
+### Portability
+A type is *portable to* a subgraph if it has a `@csdl_key` for that subgraph.
+
+### Free / Bound fields
+
+Free fields can be resolved by any subgraph, bound fields can only be resolved by the subgraph they're bound to.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
+
+```graphql
+#::[Using `csdl_Graph` to define endpoints]
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+```
+
+The `csdl_Graph` enum is used as input to csdl directives which link fields and types to subgraphs.
+
+## Provided Types
+
+Valid CSDL documents **must** include the declarations of any CSDL types they use.
+
+### csdl_Never
+```graphql
+#::[Declaration of csdl_Never]
+type csdl_Never {}
+```
+
+The `csdl_Never` type is used to indicate the type of a field which **cannot be resolved**. This is typically due to a composition error. Producers **should** provide more specific error information with a `@csdl_Error` directive on the relevant field.
+
+```graphql
+#::[Using `csdl_Never` and `@csdl_error` to mark a field error]
+type SomeType {
+  someField: csdl_Never
+    @csdl_error(message: "someField defined with conflicting types: Int, String")
+}
+```
+
+## Scalars
+
+Valid CSDL documents **must** include the declarations of any CSDL scalars they use. Producers **should** include the `@specifiedBy` directive, but it is not mandatory.
+
+### `csdl_SelectionSet`
+
+```graphql
+#::[`scalar csdl_SelectionSet` declaration]
+scalar csdl_SelectionSet
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_selectionset")
+```
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+All legal selection set features are allowed. That includes alises, field arguments, directives, and inline fragments. Named fragment spreads are syntactically valid, but meaningless, as there is nowhere to define named fragments; future versions of this spec may change this.
+
+### `csdl_URL`
```

Comment:
Of course now that I actually read the GraphQL spec I see that it suggests `Url` as a possible scalar name. Heh. Well, I guess it can go either way.

Reply:
I'm guessing they're using a capitalization convention like [the MS one](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions), where they capitalize all letters for acronyms/initialisms that are two letters long, but capitalize just the first letter if longer. I'd be fine with using that one.